### PR TITLE
[dart/en] Update Guide and add more features

### PR DESCRIPTION
**Pull request by ram231**
- [x] I solemnly swear that this is all original content of which I am the original author(I'm just a contributor)
- [x] Pull request title is prepended with `[language/lang-code]`
- [x] Pull request touches only one file (or a set of logically related files with similar changes made)
- [x] Content changes are aimed at *intermediate to experienced programmers* (this is a poor format for explaining fundamental programming concepts)
- [x] If you've changed any part of the YAML Frontmatter, make sure it is formatted according to [CONTRIBUTING.md](https://github.com/adambard/learnxinyminutes-docs/blob/master/CONTRIBUTING.markdown)
  - [x] Yes, I have double-checked quotes and field names!

Some guides have missing features that is needed to understand the dart specifications.
Added the following:
- Generics
- List
- dynamic keyword
- num keyword
- Null aware operator

### CODE REVIEW

**Comment by divayprakash** [comment]
@ram231 Any reason for closing this? Sorry about the delay in merging, as you can see we're currently working through a large backlog of PRs.

**Comment by divayprakash** [comment]
I've resolved the merge conflict, do let me know if this is ready to merge.

File: dart.html.markdown [hunk 1]
```diff
@@ -2,28 +2,72 @@
...
+  - ["Vince Ramces Oliveros", "https://github.com/ram231"]
 ---
 
-Dart is a newcomer into the realm of programming languages.
-It borrows a lot from other mainstream languages, having as a goal not to deviate too much from
-its JavaScript sibling. Like JavaScript, Dart aims for great browser integration.
+**Dart** is a single threaded, general puprose programming languages.
+It borrows a lot from other mainstream languages.
+It supports Streams, Futures(known as Promises in JavaScript), Generics, First-class functions(closures) and static type checking.
+Dart can run in any platform such as Web, CLI,Desktop, Mobile and IoT devices.
```
Comment (megatux) [inline dart.html.markdown]: add space after "CLI,"
Also, maybe we could add something about running on a VM or native (ARM and X86)

**Comment by divayprakash** [comment]
@megatux I'm gonna go ahead and merge this in as is for now. We can look at any other issues further down the line.

**Comment by eximius313** [comment]
@ram231, I just started learning Dart, so I copied your code, pasted it in VS Code and... it does not even compile...

I've noticed that in import you changed to `math`, but in the code it is still `DM`.
Also there is something wrong with assigning numbers:
```
  myInt = myNumDouble; //Error
  myInt = myDouble; //Error
  myInt = myNumInt; //valid

  myDouble = myInt; //error
  myDouble = myNumInt; //valid
  myDouble = myNumDouble; //valid
```
^ this whole section doesn't compile etither...
There are also problems with nulls and default values.

As I'm a beginner so I don't know what's wrong.
Are you able to verify that the code compiles?

Thank you in advance

**Pull request merged**